.github/workflows/backend.yml

```diff
@@ -0,0 +1,125 @@
+name: Backend Checks
+
+on:
+  push:
+    # branches:
+    #   - main
+    #   - prod-deploy
+  pull_request:
+    paths:
+      - "training/**"
+      - pyproject.toml
+      - .github/workflows
+
+jobs:
+  lint:
+    runs-on: ubuntu-22.04
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up mamba env
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v3
+        
+      - uses: mamba-org/setup-micromamba@v1
+        with:
+          micromamba-version: '1.4.5-0'
+          environment-file: training/environment.yml
+          init-shell: >-
+            bash
+          cache-environment: true
+          cache-environment-key: environment-${{ github.sha }}
+          post-cleanup: 'all'
+      
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: training/.venv

```

Comment:
Is there a way to cache the entire mamba environment instead of the poetry deps? I think /.venv is where the env used to be stored when we still used only poetry, mamba should store it's environment in a different place

Reply:
What place should that be? @dwu359

Reply:
somewhere ending in /mambaforge/envs I think. You can also run `conda config --show envs_dirs` to get the directory that way

Reply:
updated. thanks for your help!